data_store/messaging/tf_messaging.py

```diff
@@ -296,7 +296,7 @@ def _wrong_type_failure_message(self, validation_failure: WrongTypeFailure) -> M
         actual_type = self.INTERNAL_TYPE_TO_MESSAGE_FORMAT[validation_failure.actual_type]
         cell_index = self._construct_cell_index(
             table=validation_failure.table,
-            column=validation_failure.column,
+            column=str(validation_failure.column),
```

Comment:
This one I am not proud of, but it was the simplest way out. I will explain it a bit.

`validation_failure.column` can be `str | list[str]`
`_construct_cell_index` accepts only `column: str`. 

It only happens for one type of `validation_failure` to have a `column` as `list[str]`. When `column` is a list, we loop call `_construct_cell_index` for each column. 

The correct approach is to always loop call `_construct_cell_index` for each column regardless of the type of `validation_failure`. But this was perhaps enough to avoid problems for the rest of the validation failures.

Reply:
If it's a `list[str]` then stringifying it will give you something like `['A']`, which will feed into `column_letter = self.TABLE_AND_COLUMN_TO_ORIGINAL_COLUMN_LETTER[table][column]` and then throw a KeyError. IE it would be broken.

If it can ever be a `list[str]` in reality, we should handle this better otherwise we're just getting rid of a mypy error but still leaving the code broken :(

Reply:
:disappointed: fine. I will loop them. 

But there is this feeling that having both str and list of str as a type is the bigger problem. But that's much harder to fix.

Reply:
don't disagree necessarily...